src/Modules/Profile/Commands.Profile/ConnectPowerBIServiceAccount.cs

```diff
@@ -61,7 +67,29 @@ public class ConnectPowerBIServiceAccount : PowerBICmdlet
 
         public override void ExecuteCmdlet()
         {
-            var environment = this.Settings.Environments[this.Environment];
+            IPowerBIEnvironment environment = null;
+
+            // re-initialize settings with custom environments from discovery url if it exists,
+            // otherwise use existing settings.
+            if (!string.IsNullOrEmpty(this.DiscoveryUrl))
+            {
```

Comment:
Will you require customers to also specificy -Environment Custom or if they did specify that, enforce they specify -CustomEnvironment and -DiscoveryUrl

Reply:
Specifying -Environment Custom is not required. If discovery url was supplied, then CustomEnvironment is a required parameter and the presence of -Environment is not required and will be ignored.